### PR TITLE
[front] style: remove the home page background alternation

### PR DESCRIPTION
**Pull request by GresilleSiffle**
**related to** https://github.com/tournesol-app/tournesol/issues/1168

---

This PR makes the style of the home pages more easily configurable (layout and color).

The previous component `<AlternatingBackgroundColorSectionList>` is not used anymore. To replace it each home page uses its own grid to list its sections, giving the developers the control over the style of each section. The home pages can now display a uniform background or to select precise color for each of their section.

No modification has been made to the home pages style, only the "inner logic" has changed. One exception: the statistics now have a grey background instead of the original blue on the video home page.

**to-do**
- [x] completely remove the `<AlternatingBackgroundColorSectionList>` component

### CODE REVIEW

**Comment by aidanjungo** [comment]
Yes it will look better now. I think we could also remove the blue section for the recommendation part and keep the blue for small blue rectangle and the footer.

**Comment by GresilleSiffle** [comment]
> Yes it will look better now. I think we could also remove the blue section for the recommendation part and keep the blue for small blue rectangle and the footer.

Sure, it's still on my plan :+1:

I didn't do it in this PR so that we can merge sooner the refactor of the sections loop, it will ease the maintenance of all my branches related to the home page. Also I may propose different background configurations for the home page. To have enough time to experiment I'd like to separate the logic PR from the style PR.

(The backgrounds I'd like to try: uniform very light-grey color, or few stripes of grey for un-titled sections, or the original blue configuration without the alternation of blue/white sections.)

File: frontend/src/pages/home/presidentielle2022/HomePresidentielle2022.tsx [hunk 1]
```diff
@@ -1,110 +1,135 @@
...
   const { baseUrl, active } = useCurrentPoll();
 
+  const homeSectionSx = {
+    width: '100%',
+    padding: 6,
+    [theme.breakpoints.down('md')]: {
+      padding: 6,
+      px: { xs: 2, md: 6 },
+    },
+  };
```
Comment (amatissart) [inline frontend/src/pages/home/presidentielle2022/HomePresidentielle2022.tsx:35]: Isn't it equivalent to

```suggestion
  const homeSectionSx = {
    width: '100%',
    padding: 6,
    px: { xs: 2, md: 6 },
  };
```

File: frontend/src/pages/home/videos/HomeVideos.tsx [hunk 2]
```diff
@@ -16,13 +23,24 @@ import { DEFAULT_POLL_STATS, getPollStats } from 'src/utils/api/stats';
...
   const [stats, setStats] = useState<PollStats>(DEFAULT_POLL_STATS);
 
+  const homeSectionSx = {
+    width: '100%',
+    padding: 6,
+    [theme.breakpoints.down('md')]: {
+      padding: 6,
+      px: { xs: 2, md: 6 },
+    },
+  };
```
Comment (amatissart) [inline frontend/src/pages/home/videos/HomeVideos.tsx:42]: same here

```suggestion
  const homeSectionSx = {
    width: '100%',
    padding: 6,
    px: { xs: 2, md: 6 },
  };
```

File: frontend/src/pages/home/videos/HomeVideos.tsx [hunk 3]
```diff
@@ -43,86 +61,103 @@ const HomeVideosPage = () => {
   }, [pollName, showWarningAlert, t]);
 
   return (
-    <AlternatingBackgroundColorSectionList lastItemIsPrimary>
-      <TitleSection title={t('home.collaborativeContentRecommendations')}>
-        <Typography paragraph fontSize="1.1em">
-          {t('home.tournesolIsAParticipatoryResearchProject')}
-        </Typography>
+    <>
+      <Box padding={4} color="white" bgcolor="#1282B2">
```
Comment (amatissart) [inline frontend/src/pages/home/videos/HomeVideos.tsx:65]: Could we define this color as a constant somewhere?  
Maybe even in "theme.ts" as a new variant `background.home` ?

File: frontend/src/pages/home/presidentielle2022/HomePresidentielle2022.tsx (hunk 1 above)
Comment (GresilleSiffle) [reply frontend/src/pages/home/presidentielle2022/HomePresidentielle2022.tsx:35]: Yes probably, I kept the original code without trying to improve it. I'm going fix this.

File: frontend/src/pages/home/videos/HomeVideos.tsx (hunk 3 above)
Comment (GresilleSiffle) [reply frontend/src/pages/home/videos/HomeVideos.tsx:65]: Yes it's a good idea.

I don't know how to name the they, as this blue is used on other pages like the FAQ or About.

Maybe `background.secondary`?

Comment (GresilleSiffle) [reply frontend/src/pages/home/videos/HomeVideos.tsx]: Should be fixed by https://github.com/tournesol-app/tournesol/pull/1287/commits/6576a8f15fdae9de7c5aba798cd325961a63c301

I also updated all components using this color.

File: frontend/src/pages/home/videos/HomeVideos.tsx (hunk 2 above)
Comment (GresilleSiffle) [reply frontend/src/pages/home/videos/HomeVideos.tsx]: Fixed by https://github.com/tournesol-app/tournesol/pull/1287/commits/6576a8f15fdae9de7c5aba798cd325961a63c301

File: frontend/src/pages/home/presidentielle2022/HomePresidentielle2022.tsx (hunk 1 above)
Comment (GresilleSiffle) [reply frontend/src/pages/home/presidentielle2022/HomePresidentielle2022.tsx]: Fixed by https://github.com/tournesol-app/tournesol/pull/1287/commits/6576a8f15fdae9de7c5aba798cd325961a63c301

File: frontend/src/pages/home/videos/HomeVideos.tsx (hunk 3 above)
Comment (amatissart) [reply frontend/src/pages/home/videos/HomeVideos.tsx]: "background.secondary" sounds good to me.  
Or "background.emphatic" to express that it's a flashier color compared to the default background.

**Review by amatissart — APPROVED** [review]

File: frontend/src/pages/home/videos/HomeVideos.tsx (hunk 3 above)
Comment (GresilleSiffle) [reply frontend/src/pages/home/videos/HomeVideos.tsx]: Good, I'll use `emphatic`.

**Pull request merged**